bugsnag-android-core/src/main/java/com/bugsnag/android/CallbackState.kt

```diff
@@ -1,11 +1,12 @@
 package com.bugsnag.android
 
-import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.CopyOnWriteArrayList
 
 internal data class CallbackState(
-    val onErrorTasks: MutableCollection<OnErrorCallback> = ConcurrentLinkedQueue<OnErrorCallback>(),
-    val onBreadcrumbTasks: MutableCollection<OnBreadcrumbCallback> = ConcurrentLinkedQueue<OnBreadcrumbCallback>(),
-    val onSessionTasks: MutableCollection<OnSessionCallback> = ConcurrentLinkedQueue()
+    val onErrorTasks: MutableCollection<OnErrorCallback> = CopyOnWriteArrayList(),
+    val onBreadcrumbTasks: MutableCollection<OnBreadcrumbCallback> = CopyOnWriteArrayList(),
+    val onSessionTasks: MutableCollection<OnSessionCallback> = CopyOnWriteArrayList(),
+    val onSendTasks: MutableCollection<OnSendCallback> = CopyOnWriteArrayList()
```

Comment:
why is this better as CoW lists?

Reply:
CoW lists will typically have slightly better space / time performance for this use-case (haven't directly benchmarked for our use case). Specifically: we tend to add / remove from the list very few times, but iterate a very large numbers of times.

Happy to change it back if that's desirable.